### PR TITLE
NAVAND-1103: add Intersection and Junction

### PR DESCRIPTION
**Pull request by dzinad**

### CODE REVIEW

File: changelog/unreleased/features/6943.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+- Added `RoadObjectType.IC` and `RoadObjectType.JCT` for interchanges and junctions and the corresponding classes `Interchange` and `Junction`.
+- Limited number of reported alternatives to 2.
```
Comment (VysotskiVadim) [inline changelog/unreleased/features/6943.md:2]:
```suggestion
- Limited number of reported alternatives to 2, i.e. the `alternatives` parameter in `RouteAlternativesObserver#onRouteAlternatives` has maximum size of 2.
```

File: changelog/unreleased/features/6943.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
+- Sped up onboard routing cancellation.
+- Fixed null pointer dereference.
```
Comment (VysotskiVadim) [inline changelog/unreleased/features/6943.md:4]: I'm not the SDK user, but I don't understand what it means. Do we need it in the changelog?
```suggestion
```

File: libnavigation-base/src/main/java/com/mapbox/navigation/base/trip/model/roadobject/LocalizedString.kt [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+ * A wrapper that contains a string in a particular language and the info
+ * about the corresponding language.
+ *
+ * @param language language which the string is in
+ * @param value original string value
+ */
+class LocalizedString internal constructor(
+    val language: String,
+    val value: String,
+) {
```
Comment (VysotskiVadim) [inline libnavigation-base/src/main/java/com/mapbox/navigation/base/trip/model/roadobject/LocalizedString.kt:13]: What is the use case for this class, i.e. how are users supposed to use it and why we don't have the same structure for the other road objects or other stuff? Is it something applicable only for new route objects or we will be adopting it for other cases?

**Review by VysotskiVadim — APPROVED** [review]
What do you think about adding an instrumentation tests(or updating existing) for new route object to make sure that integration between us and NN works correct? BTW, I think it's okay to add it in the followup PR.
@dzinad , do you need any help with manual testing? Please reach me if you need.

**Comment by dzinad** [comment]
> @dzinad , do you need any help with manual testing?

Manual testing was OK. Yeah, I'll add instrumentation tests separately.

File: changelog/unreleased/features/6943.md (hunk 2 above)
Comment (dzinad) [reply changelog/unreleased/features/6943.md:4]: Well, I think it was causing a native crash, so why not mention it.

Comment (LukasPaczos) [reply changelog/unreleased/features/6943.md:4]: This gives no context as to which crash and what was causing it. I agree that we should either provide more info (in this case repro steps) or skip it.

File: changelog/unreleased/features/6943.md [hunk 4]
```diff
@@ -0,0 +1,5 @@
...
+- Fixed unexpected NaN values in location data.
```
Comment (LukasPaczos) [inline changelog/unreleased/features/6943.md:5]:
```suggestion
- Fixed unexpected NaN values in location data.
- Increased terminal offboard route request timeout from 4 to 15 seconds.
```

Comment (LukasPaczos) [inline changelog/unreleased/features/6943.md:5]: This should all be in improvements changelog section, not features.

File: libnavigation-base/src/main/java/com/mapbox/navigation/base/trip/model/roadobject/LocalizedString.kt (hunk 3 above)
Comment (dzinad) [reply libnavigation-base/src/main/java/com/mapbox/navigation/base/trip/model/roadobject/LocalizedString.kt:13]: The names for Junction/Interchange come as a map of language-value. This class represents the pair language-value. We used a similar approach for `EHorizonEdgeMetadata#names`, but there we have more than language and string so it's another class (`RoadComponent`).

File: changelog/unreleased/features/6943.md (hunk 4 above)
Comment (dzinad) [reply changelog/unreleased/features/6943.md]: Oops, I forgot we had different directories...

**Pull request merged**